### ISSUE TITLE
Warning message at boot when asked for LUKS password

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#13499]
**systemd version the issue has been seen with**

sys-apps/systemd-243_rc2-r1

**Used distribution**

Gentoo Linux.

**Expected behaviour you didn't see**

Clean password prompt at boot without any warning messages.

**Unexpected behaviour you saw**

Warning message at boot when asked for LUKS password.

I have the following relevant warning messages in `journalctl`:

```
Sep 08 12:25:00 desktop dracut-initqueue[264]: Warning: The unit file, source configuration file or drop-ins of systemd-cryptsetup@luks\x2d59f57214\x2d03b2\x2d4f0a\x2d889d\x2de3c28915695d.service changed on disk. Run 'systemctl daemon-reload' to reload units.
Sep 08 12:27:34 desktop dracut-initqueue[264]: Warning: The unit file, source configuration file or drop-ins of systemd-cryptsetup@luks\x2d70b2f254\x2d2909\x2d4506\x2da1f4\x2d99a4614b8944.service changed on disk. Run 'systemctl daemon-reload' to reload units.
```

**Steps to reproduce the problem**

Upgrade sys-apps/systemd from version 242-r7 to version 243_rc2-r1.
Run `dracut -f`.
Reboot.

**Other info**

systemd-242-r7 works fine without any problems.

See also issue #6611.

I have `sys-kernel/dracut-049-r2`.

`/etc/crypttab` exists and is a symlink to an empty file.

**Comment by fturco** [comment]
I just finished doing a `git bisect` on systemd between tags `v242` and `v243-rc1`:

    e8630e695232bdfcd16b55f3faafb4329c961104 is the first bad commit

**Comment by keszybz** [comment]
Is the issue fixed with 5a1c1b534f or later?

**Comment by fturco** [comment]
I just successfully tried commit 5a1c1b5. The warning message at boot has disappeared.

Also:

```
# journalctl -b -u dracut-initqueue
-- Logs begin at Sun 2019-10-20 05:43:02 CEST, end at Wed 2019-11-06 17:17:08 CET. --
Nov 06 17:15:46 desktop systemd[1]: Starting dracut initqueue hook...
Nov 06 17:15:57 desktop systemd[1]: Started dracut initqueue hook.
Nov 06 17:15:59 desktop systemd[1]: dracut-initqueue.service: Succeeded.
Nov 06 17:15:59 desktop systemd[1]: Stopped dracut initqueue hook.
```

Thank you for having fixed this issue!
